mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/task/SolveMessageInconsistenciesServiceTest.java

```diff
@@ -0,0 +1,230 @@
+    }
+
+    @Test
+    void fixMessageInconsistenciesShouldReturnPartialWhenPartialError() {
+        imapUidDAO.insert(MESSAGE_1).block();
+        imapUidDAO.insert(MESSAGE_2).block();
+
+        doReturn(Mono.error(new RuntimeException())).when(messageIdDAO).insert(MESSAGE_1);
+
+        assertThat(testee.fixMessageInconsistencies().block())
+            .isEqualTo(Task.Result.PARTIAL);
+    }
+
+    @Test
+    void fixMessageInconsistenciesShouldResolveSuccessPartially() {
+        imapUidDAO.insert(MESSAGE_1).block();
+        imapUidDAO.insert(MESSAGE_2).block();
+
+        doReturn(Mono.error(new RuntimeException())).when(messageIdDAO).insert(MESSAGE_1);
+
+        testee.fixMessageInconsistencies().block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(imapUidDAO.retrieve(MESSAGE_ID_2, Optional.of(MAILBOX_ID)).collectList().block())
+                .containsExactly(MESSAGE_2);
+            softly.assertThat(messageIdDAO.retrieve(MAILBOX_ID, MESSAGE_UID_2).block().get())
+                .isEqualTo(MESSAGE_2);
+        });
+    }
+
+    @Test
+    void fixMessageInconsistenciesShouldReturnCompletedWhenInconsistentFlags() {
+        imapUidDAO.insert(MESSAGE_1).block();
+        messageIdDAO.insert(MESSAGE_1_WITH_SEEN_FLAG).block();
+
+        assertThat(testee.fixMessageInconsistencies().block())
+            .isEqualTo(Task.Result.COMPLETED);
+    }
+
+    @Test
+    void fixMessageInconsistenciesShouldResolveInconsistentFlags() {
+        imapUidDAO.insert(MESSAGE_1).block();
+        messageIdDAO.insert(MESSAGE_1_WITH_SEEN_FLAG).block();
+
+        testee.fixMessageInconsistencies().block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(imapUidDAO.retrieve(MESSAGE_ID_1, Optional.of(MAILBOX_ID)).collectList().block())
+                .containsExactly(MESSAGE_1);
+            softly.assertThat(messageIdDAO.retrieve(MAILBOX_ID, MESSAGE_UID_1).block().get())
+                .isEqualTo(MESSAGE_1);
+        });
+    }
+
+    @Test
+    void fixMessageInconsistenciesShouldReturnCompletedWhenInconsistentModSeq() {
+        imapUidDAO.insert(MESSAGE_1).block();
+        messageIdDAO.insert(MESSAGE_1_WITH_MOD_SEQ_2).block();
+
+        assertThat(testee.fixMessageInconsistencies().block())
+            .isEqualTo(Task.Result.COMPLETED);
+    }
+
+    @Test
+    void fixMessageInconsistenciesShouldResolveInconsistentModSeq() {
+        imapUidDAO.insert(MESSAGE_1).block();
+        messageIdDAO.insert(MESSAGE_1_WITH_MOD_SEQ_2).block();
+
+        testee.fixMessageInconsistencies().block();
+
+        SoftAssertions.assertSoftly(softly -> {
+            softly.assertThat(imapUidDAO.retrieve(MESSAGE_ID_1, Optional.of(MAILBOX_ID)).collectList().block())
+                .containsExactly(MESSAGE_1);
+            softly.assertThat(messageIdDAO.retrieve(MAILBOX_ID, MESSAGE_UID_1).block().get())
+                .isEqualTo(MESSAGE_1);
+        });
+    }
+}
```

Comment:
Please add a test showing me that if an entry is in imapUidDAO and not in messageIdTable, it is added to messageIdTable.

Please add a test showing me that if an entry is in messageIdTable and not in imapUidDAO, it is removed from messageIdTable.